ethcore/sync/src/chain/mod.rs

```diff
@@ -335,6 +339,8 @@ pub struct PeerInfo {
 	network_id: u64,
 	/// Peer best block hash
 	latest_hash: H256,
+	/// Unpropagated tx pool hashes
+	unsent_pooled_hashes: Option<H256FastSet>,
```

Comment:
What is the point of `Option` here? Is `None` different from the hash set being empty, semantically? I mean to say: isn't it unlikely we'll be running with this set `None` during normal operation, and if so there's not much of a space saving?